packages/react/readme.md

```diff
@@ -1,4 +1,90 @@
-
 # Hops React
 
-T.B.D. (To Be Documented)
+hops-react works in tandem with [hops-cli](https://github.com/xing/hops/blob/master/packages/cli/index.js) and [hops-config](https://github.com/xing/hops/blob/master/packages/redux/index.js) to make an integrated solution for universal ("isomorphic") rendering using React. It provides a minimal API and hides the tricky bits of setting up React for such use-cases.
+
+Out of the box, hops-react additionally supports [React Router](https://github.com/ReactTraining/react-router) and [React Helmet](https://github.com/nfl/react-helmet).
+
+# Installation
+``` bash
+npm install --save react react-dom react-helmet react-router react-router-dom hops-react
+```
+
+# API
+## `render(reactElement, context)`
+`render()` is the main hops-react API. Its return value differs depending on the environment: in a browser context, it returns a plain function handling mounting of the app and hot module replacement. In a Node.js context, it returns an Express style middleware function. In both cases, its return value is meant to be the default export of your application's main entry file.
+
+`render()` accepts two arguments: `reactElement`, your React application's root element, and `context`, an optional hops-react rendering context instance.
+
+``` js
+import React from 'react';
+import { render } from 'hops-react';
+
+const App = () => (<h1>Hello World!</h1>);
+
+export default render(<App />)
+```
+
+If you pass an object literal instead of a context instance, `render()` itself creates a context using said object as options.
+
+## `createContext(options)`
+`createContext()` generates a rendering context containing most of the actual implementation used by `render`. It takes a couple of options:
+
+| Field | Type | Default | Description |
+|-------|------|---------|-------------|
+| mountpoint | String | '#main' | querySelector identifying the root DOM node |
+| template | Function | defaultTemplate | template function supporting all React Helmet and hops-react features |
+
+## `Context(options)`
+If you, for whatever strange reason, prefer object-oriented syntax, you can alternatively import the `Context` constructor and use it instead of `createContext()`.
+
+If you need or want to change hops-react behavior, providing a custom Context is the way to go. To do that, you can simply use `Context.extend()`, creating a bespoke context constructor and factory function:
```

Comment:
How about documenting all of the overridable `Context` methods? (`bootstrap`, `getTemplateDate` etc)